pkg/kubectl/rolling_updater.go

```diff
@@ -67,16 +59,17 @@ type RollingUpdaterConfig struct {
 	// CleanupPolicy defines the cleanup action to take after the deployment is
 	// complete.
 	CleanupPolicy RollingUpdaterCleanupPolicy
-	// UpdateAcceptor is given a chance to accept the new controller after each
-	// scale-up operation. If the controller is accepted, updates continue; if
-	// the controller is rejected, the update will fail immediately.
-	UpdateAcceptor UpdateAcceptor
-	// UpdatePercent is optional; if specified, the amount of replicas scaled up
-	// and down each interval will be computed as a percentage of the desired
-	// replicas for the new RC. If UpdatePercent is nil, one replica will be
-	// scaled up and down each interval. If UpdatePercent is negative, the order
-	// of scaling will be down/up instead of up/down.
+	// UpdatePercent is an optional percentage of replicas to scale up each
+	// interval and is used to compute the minimum pod availability during the
+	// update. If nil, a fixed value of 1 will be used for the scaling increment
+	// and minimum pod availabity. If a negative value is provided, the scaling
+	// direction will be down/up instead of up/down.
 	UpdatePercent *int
```

Comment:
I'm using A/B here to mean having A and B sides of a physical deployment (a
deployment shard).  Not an A/B test.

On Fri, Aug 7, 2015 at 11:33 AM, Dan Mace <notifications@github.com> wrote:

> In pkg/kubectl/rolling_updater.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11942#discussion_r36530622>
> :
>
> > @@ -67,16 +59,17 @@ type RollingUpdaterConfig struct {
> >  	// CleanupPolicy defines the cleanup action to take after the deployment is
> >  	// complete.
> >  	CleanupPolicy RollingUpdaterCleanupPolicy
> > -	// UpdateAcceptor is given a chance to accept the new controller after each
> > -	// scale-up operation. If the controller is accepted, updates continue; if
> > -	// the controller is rejected, the update will fail immediately.
> > -	UpdateAcceptor UpdateAcceptor
> > -	// UpdatePercent is optional; if specified, the amount of replicas scaled up
> > -	// and down each interval will be computed as a percentage of the desired
> > -	// replicas for the new RC. If UpdatePercent is nil, one replica will be
> > -	// scaled up and down each interval. If UpdatePercent is negative, the order
> > -	// of scaling will be down/up instead of up/down.
> > +	// UpdatePercent is an optional percentage of replicas to scale up each
> > +	// interval and is used to compute the minimum pod availability during the
> > +	// update. If nil, a fixed value of 1 will be used for the scaling increment
> > +	// and minimum pod availabity. If a negative value is provided, the scaling
> > +	// direction will be down/up instead of up/down.
> >  	UpdatePercent *int
>
> There's still ingoing discussion about how to build this API to best
> capture the user's intent. Holding off on changing this until we achieve
> consensus in #12236
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12236>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11942/files#r36530622>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift